Helper/Api/Products.php

```diff
@@ -1104,28 +1058,32 @@ private function processImages(array $flat_product, Product $target, bool $shoul
     {
         $galleryImages = $target->getMediaGalleryEntries();
         $existingImagesArray = [];
-        foreach ($galleryImages as $entryId => $image) {
-            $mediaGalleryImage = $target->getMediaGalleryImages()->getItemById($image->getId());
-            if ($mediaGalleryImage) {
-                $skImageId = $mediaGalleryImage->getStorekeeperImageId();
-                $skImageIds = array_column($flat_product['product_images'], 'id');
-                /**
-                 * Look for storekeeper image id of current gallery image in array of images from SK backoffice
-                 * if current product does not match id or id is missing - remove gallery image
-                 */
-                if (
-                    !in_array($skImageId, $skImageIds)
-                    && $this->productExportManager->isImageFormatAllowed($mediaGalleryImage->getPath())
-                ) {
-                    $shouldUpdate = true;
-                } else {
-                    $existingImagesArray[$skImageId] = $image;
+        if (array_key_exists('product_images', $flat_product)) {
```

Comment:
we do not remove the images anymore when all images are removed from storekeeper?